middlewares/inputValidation.js

```diff
@@ -45,6 +46,93 @@ const validate = {
       }
       return next();
     }
-  ]
+  ],
+  createProfileValidate: [
+    check('avatar')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Avatar is required')
+      .isURL()
+      .withMessage('Avatar is not a valid URL, please input a valid URL'),
+    check('bio')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Bio is required')
+      .not()
+      .isInt()
+      .withMessage('Bio is not a valid string, please input a valid string'),
+    (req, res, next) => {
+      const errors = validationResult(req);
+      const errorMessage = [];
+      if (!errors.isEmpty()) {
+        errors.array({ onlyFirstError: true }).forEach((err) => {
+          errorMessage.push(err.msg);
+        });
+        return res.status(400).json({
+          errors: errorMessage,
+        });
+      }
+      return next();
+    },
+  ],
+  editProfileValidate: [
+    check('avatar')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Avatar is required')
+      .isURL()
+      .withMessage('Avatar is not a valid URL, please input a valid URL'),
+    check('bio')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Bio is required')
+      .not()
+      .isInt()
+      .withMessage('Bio is not a valid string, please input a valid string'),
+    check('firstName')
+      .isAlpha()
+      .trim()
+      .withMessage('FirstName is not a valid String, please input a valid string'),
+    check('lastName')
+      .isAlpha()
+      .trim()
+      .withMessage('LastName is not a valid string, please input a valid string'),
+    (req, res, next) => {
+      const errors = validationResult(req);
+      const errorMessage = [];
+      if (!errors.isEmpty()) {
+        errors.array({ onlyFirstError: true }).forEach((err) => {
+          errorMessage.push(err.msg);
+        });
+        return res.status(400).json({
+          errors: errorMessage,
+        });
+      }
+      return next();
+    },
+  ],
+  validateId: [
+    param('id')
+      .custom(async (id) => {
+        const isExist = await models.User.findOne({ where: { id } });
+        if (!isExist) {
+          throw new Error('No User with the specified ID was found');
+        }
+        return true;
+      }),
+    (req, res, next) => {
+      const errors = validationResult(req);
+      const errorMessage = [];
```

Comment:
I remember we said something about returning validation errors as an object instead of an array. Does it not apply here?